.circleci/config.yml

```diff
@@ -351,12 +351,34 @@ defaults:
 
   - steps_install_dependencies_osx: &steps_install_dependencies_osx
       steps:
-        # FIXME: We used to cache dependencies on macOS but now it takes longer than just installing
-        # them each time. See https://github.com/ethereum/solidity/issues/12925.
+        - run:
+            name: Cleaning Cellar directory
+            command: |
+              sudo rm -rf /usr/local/Cellar
+              sudo mkdir -p /usr/local/Cellar
+              sudo chown -R distiller:admin /usr/local/Cellar
+        - restore_cache:
+            keys:
+              - dependencies-osx-v1-{{ arch }}-{{ checksum ".circleci/osx_install_dependencies.sh" }}
+        - attach_workspace:
+            at: .
+          # DO NOT EDIT between here and save_cache, but rather edit ./circleci/osx_install_dependencies.sh
+          # WARNING! If you do edit anything here instead, remember to invalidate the cache manually.
         - run:
             name: Install build dependencies
             command: ./.circleci/osx_install_dependencies.sh
 
+        - save_cache:
+            key: dependencies-osx-v1-{{ arch }}-{{ checksum ".circleci/osx_install_dependencies.sh" }}
+            paths:
+              - /usr/local/bin
+              - /usr/local/lib
+              - /usr/local/include
+              - /usr/local/opt
+              - /usr/local/share
+              - /usr/local/var
```

Comment:
Just realized that homebrew install more things that if not cached may indeed break the image. You can check it with `brew doctor` command. 
```
Warning: Some installed formulae are missing dependencies.
You should `brew install` the missing dependencies:
  brew install ca-certificates gettext gmp icu4c libidn2 libunistring lz4 openssl@1.1 xz zstd

Run `brew missing` for more details.
static:Cellar distiller$ brew missing
boost: icu4c xz lz4 zstd
coreutils: gmp
wget: gettext libunistring libidn2 ca-certificates openssl@1.1
```

Reply:
So far, the only way that I could make it work was not deleting the whole `Cellar` directory.

Reply:
Yeah, I added that only for testing, attempting to fix the errors reported by the `brew doctor`. Upon further look, it turned out that the important directory there is only `/usr/local/var/homebrew` which contains symlinks to the respective files under the `Cellar` directory.